src/components/ProductHero.tsx

```diff
@@ -0,0 +1,66 @@
+import { FunctionComponent, ReactNode, useEffect, useRef } from 'react'
+
+import { Background, HubSpotForm } from '@components'
+
+interface ProductHero extends Background {
+    product: 'code search' | 'batch changes' | 'code insights'
+    title: string | ReactNode
+    description: string
+    displayUnderNav?: boolean
+}
+
+export const ProductHero: FunctionComponent<Omit<ProductHero, 'className' | 'children' | 'illustration'>> = ({
```

Comment:
Can you explain why you're omitting some properties on here? `Omit` is a bit new to me

Reply:
Yeah for sure! It's because we don't want to allow those props in this component since they're not used directly and there's no need to do prop drilling. We also wouldn't want to do unnecessary prop drilling here. It's better to be more explicit about your props in your component composition than to allow all the things.

Reply:
Also, this component may or may not be consolidated into a global Hero component. I think there needs to be some more consolidation in our DLS so we can have a clear path about props and prevent overcomplicating components.

Reply:
Yea I think this component can be used for all heros too! -- On the omission, am I missing something, because `ProductHero` interface doesn't have `'className' | 'children' | 'illustration'` properties. So is this necessary?

Reply:
The `ProductHero` interface extends the `Background` interface and then adds a few properties that are specific to the Product Hero 🤓

Reply:
Oh geez, missed that part 🤦🏻‍♀️  - thanks for explaining!